component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,456 @@
+        } catch (Exception e) {
+            throw new SiddhiAppRuntimeException("Exception occurred when starting server to monitor "
+                    + listeningUri + ".", e);
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        if (monitor != null) {
+            try {
+                monitor.stop();
+                initialMap.clear();
+                log.debug("Directory monitoring has been stopped for folder : " + listeningUri + " .");
+            } catch (Exception e) {
+                log.error("Exception occurred when stopping server  while monitoring " + listeningUri + " .", e);
+            }
+        }
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public void pause() {
+        if (monitor != null) {
+            threadList.forEach(ThreadOnContinuousChange::pause);
+            log.debug("Directory monitoring has been paused for folder : " + listeningUri + " .");
+        }
+    }
+
+    @Override
+    public void resume() {
+        if (monitor != null) {
+            threadList.forEach(ThreadOnContinuousChange::resume);
+            log.debug("Directory monitoring has been resumed for folder : " + listeningUri + " .");
+        }
+    }
+
+    /**
+     * Implementation of SubThread
+     */
+    public class ThreadOnContinuousChange implements Runnable {
+        String status;
+        private final File file;
+        private volatile boolean paused;
+        private ReentrantLock lock;
+        private Condition condition;
+        private volatile boolean inactive;
+
+        ThreadOnContinuousChange(File file) {
+            this.file = file;
+            inactive = false;
+            lock = new ReentrantLock();
+            condition = lock.newCondition();
+        }
+
+        @Override
+        public void run() {
+            while (!inactive) {
+                if (paused) {
+                    lock.lock();
+                    try {
+                        condition.await();
+                    } catch (InterruptedException ie) {
+                        Thread.currentThread().interrupt();
+                    } finally {
+                        lock.unlock();
+                    }
+                }
+                long recentFileSize = recentFileSizeMap.get(file.getAbsolutePath());
+                if (recentFileSize < file.length()) {
+                    status = STATUS_PROCESS;
+                    recentFileSizeMap.put(file.getAbsolutePath(), file.length());
+                    ThreadOnContinuousChange threadOnContinuousChange = new ThreadOnContinuousChange(file);
+                    siddhiAppContext.getScheduledExecutorService().schedule(threadOnContinuousChange,
+                            monitoringInterval, TimeUnit.MILLISECONDS);
+                    threadList.add(threadOnContinuousChange);
+                } else {
+                    status = STATUS_DONE;
+                    recentFileSizeMap.remove(file.getAbsolutePath());
```

Comment:
After changing the threadList to a map, the specific thread object should be removed from here as well.

Reply:
fixed in 3f9a2a0